### PR TITLE
device-simulator: integrate with current iotivity-lite

### PR DESCRIPTION
**Pull request by Danielius1922**
- clone and compile current iotivity-lite repository
- fix tests after changes in iotivity-lite

### CODE REVIEW

File: device-simulator/Dockerfile [hunk 1]
```diff
@@ -1,14 +1,21 @@
...
-RUN git clone --recursive -b plgd.dev https://github.com/iotivity/iotivity-lite.git
-COPY ./src/ /iotivity-lite/
-RUN make -C /iotivity-lite/port/linux CLOUD=1 SECURE=1 OSCORE=0 DEBUG=1 MNT=1 ASAN=0 cloud_server
+  apt-get install -y bash curl gdb git-core g++ ca-certificates patch make --no-install-recommends
+ARG IOTIVITY_COMMIT=c1ae1d2cb4afc629dfa60d4ac8b613147fb04ca4
+ARG GTEST_COMMIT=c99458533a9b4c743ed51537e25989ea55944908
+ARG MBEDTLS_COMMIT=d81c11b8ab61fd5b2da8133aa73c5fe33a0633eb
+ARG TINYCBOR_COMMIT=70aba6ba51881e5b8d108c105a17ed9cdee6bc30
+RUN git clone --recursive https://github.com/iotivity/iotivity-lite.git && \
+  cd /iotivity-lite && git checkout $IOTIVITY_COMMIT && \
```
Comment (ondrejtomcik) [inline device-simulator/Dockerfile]: Why do you specify custom submodule commits?

File: pkg/ocf/ocf.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+const (
+	OC_IF_BASELINE = "oic.if.baseline"
+	OC_IF_LL       = "oic.if.ll"     // links list
+	OC_IF_B        = "oic.if.b"      // batch
+	OC_IF_R        = "oic.if.r"      // read-only
+	OC_IF_RW       = "oic.if.rw"     // read-write
+	OC_IF_A        = "oic.if.a"      // actuator
+	OC_IF_S        = "oic.if.s"      // sensor
+	OC_IF_CREATE   = "oic.if.create" // create
```
Comment (jkralik) [inline pkg/ocf/ocf.go]: I think we can move these constants to https://github.com/plgd-dev/device/tree/main/schema as an interfaces package

File: pkg/ocf/ocf.go [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+const (
+	OC_RT_P   = "oic.wk.p"   // platform
+	OC_RT_D   = "oic.wk.d"   // device
+	OC_RT_CON = "oic.wk.con" // configuration
+	OC_RT_COL = "oic.wk.col" // collection
```
Comment (jkralik) [inline pkg/ocf/ocf.go]: I think there are defined in `https://github.com/plgd-dev/device/tree/main/schema`. If it is not then we can define them.

File: pkg/ocf/ocf.go [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+	OC_RT_DEVICE_CLOUD = "oic.d.cloudDevice"
```
Comment (jkralik) [inline pkg/ocf/ocf.go]: this test resource type so move it to `hub/test/config`

File: pkg/ocf/ocf.go [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+
+	OC_RT_RESOURCE_SWITCH = "oic.r.switch.binary"
```
Comment (jkralik) [inline pkg/ocf/ocf.go]: this could be moved to https://github.com/plgd-dev/device/tree/main/schema or  `hub/test/config`

**Review by jkralik — APPROVED** [review]

**Pull request merged**